packages/core/src/testInteg/notifications/notifications.test.ts

```diff
@@ -0,0 +1,114 @@
+            activeExtensions: activeExtensions,
+        }
+
+        return new RuleEngine(ruleContext)
+    }
+
+    beforeEach(async function () {
+        panelNode = NotificationsNode.instance
+        // fetch from test host file folder
+        fetcher = new RemoteFetcher(
+            'https://idetoolkits-hostedfiles.amazonaws.com/Notifications/integ/VSCode/startup/1.x.json',
+            'https://idetoolkits-hostedfiles.amazonaws.com/Notifications/integ/VSCode/emergency/1.x.json'
+        )
+        controller = new NotificationsController(panelNode, fetcher)
+    })
+
+    // Clear all global states after each test
+    afterEach(async function () {
+        await globals.globalState.update(controller.storageKey, undefined)
+    })
+
+    it('Receive notifications polling from endpoint', async function () {
+        const engine = getEngine([VSCODE_EXTENSION_ID.amazonq], true)
+
+        await controller.pollForStartUp(engine)
+        await controller.pollForEmergencies(engine)
+
+        // Verify that notifications are stored in the state
+        const state = globals.globalState.get<NotificationsState>(controller.storageKey)
+        assert.ok(state, 'State should be defined')
+        assert.ok(state.startUp.payload?.notifications, 'StartUp received')
+        assert.strictEqual(
+            state?.startUp.payload?.notifications.length,
+            2,
+            'There should be 2 startup notifications stored'
+        )
+        assert.ok(state?.emergency.payload?.notifications, 'Emergency received')
+        assert.strictEqual(
+            state?.emergency.payload?.notifications.length,
+            2,
+            'There should be 2 emergency notifications stored'
+        )
+    })
+
+    it('Display notification according to criterias', async function () {
+        const engine = getEngine([VSCODE_EXTENSION_ID.amazonq], false)
+        await controller.pollForEmergencies(engine)
+
+        // Verify that only the not authed notification is displayed
+        const displayedNotifications = panelNode.emergencyNotifications
+        assert.strictEqual(displayedNotifications.length, 1, 'Only one notification displayed"')
+        assert.strictEqual(
+            displayedNotifications[0].id,
+            'emergency1',
+            'The displayed notification have the ID "emergency1"'
+        )
+    })
+
+    it('Should trigger onReceive only for newly received notifications', async function () {
+        const engine = getEngine([VSCODE_EXTENSION_ID.amazonq], true)
+        const onReceiveSpy = sinon.spy(panelNode, 'onReceiveNotifications')
+
+        // Simulate alreadying receving a notification
+        const existingNotificationID = 'startup1'
+        const state = globals.globalState.get<NotificationsState>(controller.storageKey)
+        if (state) {
+            state.newlyReceived.push(existingNotificationID)
+            await globals.globalState.update(controller.storageKey, state)
+        }
```

Comment:
This should usually work, since other tests that run before build the state object if it doesnt exist yet.
But in case the test is running standalone, you should have a default state object. Otherwise, this test will fail.